src/erc3525/extensions/slotapprovable/module.cairo

```diff
@@ -0,0 +1,159 @@
+            // [Check] Caller and operator are not null addresses
+            let caller = get_caller_address();
+            assert(!caller.is_zero(), 'ERC3525: invalid caller');
+            assert(!operator.is_zero(), 'ERC3525: invalid operator');
+
+            // [Check] Operator is not owner and caller is approved or owner
+            let unsafe_state = ERC721::unsafe_new_contract_state();
+            let owner = ERC721::ERC721Impl::owner_of(@unsafe_state, token_id);
+            assert(owner != operator, 'ERC3525: approval to owner');
+            self._assert_allowed(caller, token_id);
+            
+            // [Effect] Store approved value
+            let mut unsafe_state = ERC3525::unsafe_new_contract_state();
+            ERC3525::InternalImpl::_approve_value(ref unsafe_state, token_id, operator, value);
+        }
+
+        fn transfer_value_from(ref self: ContractState, from_token_id: u256, to_token_id: u256, to: ContractAddress, value: u256) -> u256 {
+            // [Check] caller, from token_id and transfered value are not null
+            let caller = get_caller_address();
+            assert(!caller.is_zero(), 'ERC3525: invalid caller');
+            assert(from_token_id != 0.into(), 'ERC3525: invalid from token id');
+            assert(value != 0.into(), 'ERC3525: invalid value');
+
+            // [Check] Disambiguate function call: only one of `to_token_id` and `to` must be set
+            assert(to_token_id == 0.into() || to.is_zero(), 'ERC3525: mutually excl args set');
+
+            // [Effect] Spend allowance if possible
+            self._spend_allowance(caller, from_token_id, value);
+
+            // [Effect] Transfer value to address
+            let mut unsafe_state = ERC3525::unsafe_new_contract_state();
+            if to_token_id == 0.into() {
+                return ERC3525::InternalImpl::_transfer_value_to(ref unsafe_state, from_token_id, to, value);
+            }
+            ERC3525::InternalImpl::_transfer_value_to_token(ref unsafe_state, from_token_id, to_token_id, value)
+        }
+    }
+
+    #[generate_trait]
+    impl InternalImpl of InternalTrait {
+        fn initializer(ref self: ContractState, value_decimals: u8) {
+            // [Effect] Register interfaces
+            let mut unsafe_state = SRC5::unsafe_new_contract_state();
+            SRC5::InternalImpl::register_interface(ref unsafe_state, IERC3525_SLOT_APPROVABLE_ID);
+        }
+
+        fn _spend_allowance(ref self: ContractState, spender: ContractAddress, token_id: u256, value: u256) {
+            // [Compute] Spender allowance
+            let is_approved = self._is_allowed(spender, token_id);
+            let mut unsafe_state = ERC3525::unsafe_new_contract_state();
+            let current_allowance = ERC3525::ERC3525Impl::allowance(@unsafe_state, token_id, spender);
+            let infinity : u256 = BoundedInt::max();
```

Comment:
in some constants.cairo?

Reply:
This is a const behind the scene, just imported from core lib, I don't think it makes sense to redefine it in another local const.